lib/Differentiator/DiffPlanner.cpp

```diff
@@ -537,51 +538,68 @@ namespace clad {
     if (A &&
         (A->getAnnotation().equals("D") || A->getAnnotation().equals("G") ||
          A->getAnnotation().equals("H") || A->getAnnotation().equals("J") ||
-         A->getAnnotation().equals("E") || A->getAnnotation().equals("VD"))) {
+         A->getAnnotation().equals("E"))) {
       // A call to clad::differentiate or clad::gradient was found.
       DeclRefExpr* DRE = getArgFunction(E, m_Sema);
       if (!DRE)
         return true;
       DiffRequest request{};
 
-      // Check if enzyme is requested in case of Forward Mode or reverse Mode
-      if (A->getAnnotation().equals("D") || A->getAnnotation().equals("G")) {
-        signed opts_value = FD->getTemplateSpecializationArgs()
-                                ->get(0)
-                                .getAsIntegral()
-                                .getZExtValue();
-
-        if (opts_value == -1) {
-          request.use_enzyme = true;
-        }
-      }
-
       if (A->getAnnotation().equals("D")) {
         request.Mode = DiffMode::forward;
-        llvm::APSInt derivativeOrderAPSInt
-          = FD->getTemplateSpecializationArgs()->get(0).getAsIntegral();
 
-        unsigned derivativeOrder = 1;
-        if (!request.use_enzyme)
-          derivativeOrder = derivativeOrderAPSInt.getZExtValue();
-        request.RequestedDerivativeOrder = derivativeOrder;
-      } else if (A->getAnnotation().equals("VD")) {
-        request.Mode = DiffMode::vector_forward_mode;
-        unsigned derivativeOrder = FD->getTemplateSpecializationArgs()
-                                       ->get(0)
-                                       .getAsIntegral()
-                                       .getZExtValue();
-        // currently only first order derivative is supported.
-        assert(derivativeOrder == 1 &&
-               "Only first order derivative is supported for now in vector "
-               "forward mode.");
+        unsigned const derivativeOrder = FD->getTemplateSpecializationArgs()
+                                             ->get(0)
+                                             .getAsIntegral()
+                                             .getZExtValue();
         request.RequestedDerivativeOrder = derivativeOrder;
+
+        unsigned const bitmasked_opts_value =
+            FD->getTemplateSpecializationArgs()
+                ->get(1)
+                .getAsIntegral()
+                .getZExtValue();
+        if (clad::HasOption(bitmasked_opts_value, clad::opts::use_enzyme)) {
+          request.use_enzyme = true;
+        }
```

Comment:
```suggestion
        if (clad::HasOption(bitmasked_opts_value, clad::opts::use_enzyme))
          request.use_enzyme = true;
```

I believe we could configure clang-format or clang-tidy to find these.